### PR TITLE
Add contributing file

### LINKED ISSUE #122: Add CONTRIBUTING.md
I think we're long overdue on a CONTRIBUTING.md file :) 

We can likely move a lot of things from the main README.md there too.

### PR DESCRIPTION
**Pull request by another-rex**
Fixes #122 . Not sure if install from source should be moved to contributing as well, but I'm leaning towards leaving it in README since people might have go and just want to install it from there.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+This project follows
+[Google's Open Source Community Guidelines](https://opensource.google.com/conduct/).
+
+## Contributing code
+
+### Prerequisites
+You must install:
+1. [Go](https://go.dev/) 1.18+
+2. [GoReleaser](https://goreleaser.com/) (Optional, only if you want reproducible builds)
```
Comment (oliverchang) [inline CONTRIBUTING.md:35]: This "optional" part conflicts a bit with line 33. 

Maybe move this to an optional dependencies section?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+```shell
+$ goreleaser build --rm-dist --single-target --snapshot
+```
+
+See GoReleaser [documentation](https://goreleaser.com/cmd/goreleaser_build/) for build options.
+
+You can also reproduce the downloadable builds by checking out the specific tag and running `goreleaser build`,
+using the same Go version as the one used during the actual release (see goreleaser workflows).
+
+#### Build using only `go`
```
Comment (oliverchang) [inline CONTRIBUTING.md:51]: This is probably the most common one, so let's have this first.

File: README.md [hunk 3]
```diff
@@ -61,18 +62,9 @@ go install github.com/google/osv-scanner/cmd/osv-scanner@v1
...
-
-```bash
-$ goreleaser build --rm-dist --single-target --snapshot
-```
-
-See GoReleaser [documentation](https://goreleaser.com/cmd/goreleaser_build/) for build options.
-
-You can also reproduce the downloadable builds by checking out the specific tag and running `goreleaser build`,
-using the same Go version as the one used during the actual release (see goreleaser workflows).
+See CONTRIBUTING.md file.
```
Comment (oliverchang) [inline README.md:67]: Let's linkify this.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+See GoReleaser [documentation](https://goreleaser.com/cmd/goreleaser_build/) for build options.
+
+You can also reproduce the downloadable builds by checking out the specific tag and running `goreleaser build`,
+using the same Go version as the one used during the actual release (see goreleaser workflows).
+
+### Running tests
+
+To run tests:
+```shell
+run_tests.sh
```
Comment (SSE4) [inline CONTRIBUTING.md:62]:
```suggestion
./run_tests.sh
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+### Linting
+To lint your code, run
+
+```shell
+run_lints.sh
```
Comment (SSE4) [inline CONTRIBUTING.md:69]:
```suggestion
./run_lints.sh
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+1. [Go](https://go.dev/) 1.18+
+2. [GoReleaser](https://goreleaser.com/) (Optional, only if you want reproducible builds)
+
+### Building
+
+#### Build using only `go`
+
+Run the following in the project directory:
+```shell
+$ go build ./cmd/osv-scanner/
```
Comment (SSE4) [inline CONTRIBUTING.md:43]: this command has no output, which is kinda confusing
I think it's useful to add a note it produces `./osv-scanner` as an executable
(no idea why is it directly in the source root, and not in `.gitignore`)

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+## Community Guidelines
+
+This project follows
+[Google's Open Source Community Guidelines](https://opensource.google.com/conduct/).
+
+## Contributing code
+
+### Prerequisites
+Install:
+1. [Go](https://go.dev/) 1.18+
```
Comment (SSE4) [inline CONTRIBUTING.md:34]:
```suggestion
1. [Go](https://go.dev/) 1.18+, use `go version` to check
```

ironically, it doesn't like `go --version`, `go -version` or `go -v`. for some reason, it uses weird command line syntax :thinking:  it confused me.

File: CONTRIBUTING.md (hunk 6 above)
Comment (another-rex) [reply CONTRIBUTING.md:43]: Good point, added an explanation and put /osv-scanner into .gitignore

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+### Building
+
+#### Build using only `go`
+
+Run the following in the project directory:
+```shell
+$ go build ./cmd/osv-scanner/
+```
+Outputs osv-scanner binary will be produced in the project directory.
```
Comment (oliverchang) [inline CONTRIBUTING.md:45]:
```suggestion
An osv-scanner binary will be produced in the project directory.
```

File: README.md (hunk 3 above)
Comment (oliverchang) [reply README.md:67]: Did you need to commit/push this change?

**Review by oliverchang — APPROVED** [review]

**Pull request merged**